typescript/cli/src/deploy/warp.ts

```diff
@@ -316,40 +318,47 @@ function getTokenName(token: TokenConfig) {
 function writeTokenDeploymentArtifacts(
   filePath: string,
   contracts: HyperlaneContractsMap<TokenFactories>,
-  { configMap }: DeployParams,
+  { configMap, isNft }: DeployParams,
 ) {
   const artifacts: ChainMap<{
     router: Address;
     tokenType: TokenType;
   }> = objMap(contracts, (chain, contract) => {
+    const tokenType = configMap[chain].type;
+    const contractName = isNft
+      ? hypERC721contracts[tokenType as keyof typeof hypERC721contracts]
+      : hypERC20contracts[tokenType as keyof typeof hypERC20contracts];
     return {
-      router: contract[configMap[chain].type as keyof TokenFactories].address,
-      tokenType: configMap[chain].type,
+      router: contract[contractName as keyof TokenFactories]['address'],
+      tokenType,
```

Comment:
honestly im not a fan of this different artifact shape
its another unnecessary layer of indirection

Reply:
[ContractSourceName]: address is more desirable
is this format recovered too?